app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -217,40 +220,94 @@ public static void enqueueNextOnPlayer(final Context context, final PlayQueue qu
-
-        if (index == -1) {
+        final List<AudioStream> audioStreams = info.getAudioStreams();
+        if (audioStreams.isEmpty()) {
             Toast.makeText(context, R.string.audio_streams_empty, Toast.LENGTH_SHORT).show();
             return;
         }
+        final List<AudioStream> audioStreamsForExternalPlayers = removeNonUrlAndTorrentStreams(
+                audioStreams);
+        if (audioStreamsForExternalPlayers.isEmpty()) {
+            Toast.makeText(context, R.string.no_audio_streams_available_for_external_players,
+                    Toast.LENGTH_SHORT).show();
+            return;
+        }
+        final int index = ListHelper.getDefaultAudioFormat(context,
+                audioStreamsForExternalPlayers);
 
-        final AudioStream audioStream = info.getAudioStreams().get(index);
+        final AudioStream audioStream = audioStreamsForExternalPlayers.get(index);
         playOnExternalPlayer(context, info.getName(), info.getUploaderName(), audioStream);
     }
 
-    public static void playOnExternalVideoPlayer(@NonNull final Context context,
+    public static void playOnExternalVideoPlayer(final Context context,
                                                  @NonNull final StreamInfo info) {
-        final ArrayList<VideoStream> videoStreamsList = new ArrayList<>(
-                ListHelper.getSortedStreamVideosList(context, info.getVideoStreams(), null, false,
-                        false));
-        final int index = ListHelper.getDefaultResolutionIndex(context, videoStreamsList);
-
-        if (index == -1) {
+        final List<VideoStream> videoStreams = info.getVideoStreams();
+        if (videoStreams.isEmpty()) {
             Toast.makeText(context, R.string.video_streams_empty, Toast.LENGTH_SHORT).show();
             return;
         }
+        final List<VideoStream> videoStreamsForExternalPlayers =
+                ListHelper.getSortedStreamVideosList(context,
+                        removeNonUrlAndTorrentStreams(videoStreams), null, false, false);
+        if (videoStreamsForExternalPlayers.isEmpty()) {
+            Toast.makeText(context, R.string.no_video_streams_available_for_external_players,
+                    Toast.LENGTH_SHORT).show();
+            return;
+        }
+        final int index = ListHelper.getDefaultResolutionIndex(context,
+                videoStreamsForExternalPlayers);
 
-        final VideoStream videoStream = videoStreamsList.get(index);
+        final VideoStream videoStream = videoStreamsForExternalPlayers.get(index);
         playOnExternalPlayer(context, info.getName(), info.getUploaderName(), videoStream);
     }
 
     public static void playOnExternalPlayer(@NonNull final Context context,
                                             @Nullable final String name,
                                             @Nullable final String artist,
                                             @NonNull final Stream stream) {
```

Comment:
Here, for example, `@Nullable`s do make sense, since they signal something unusual, i.e. that some arguments can even be null, but the `@NonNull`s can be removed.